File: roundRobin.js
// Initializing three arrays
const arr1 = [];
const arr2 = [];
const arr3 = [];
// Initializing a counter to keep track of the array in which the element will be pushed
let c = 1;
// Defining a function to allocate an element to an array
function allocateElement(element) {
  if (c === 1) {
    arr1.push(element);
    c = 2;
  } else if (c === 2) {
    arr2.push(element);
    c = 3;
  } else if (c === 3) {
    arr3.push(element);
    c = 1;
  }
}
// Allocating elements to the arrays using the round-robin method
allocateElement('A'); // In arr1
allocateElement('B'); // In arr2
allocateElement('C'); // In arr3
allocateElement('D'); // In arr1
allocateElement('E'); // In arr2
allocateElement('F'); // In arr3
// Output the arrays to the console to verify the results
console.log(arr1); // [ 'A', 'D' ]
console.log(arr2); // [ 'B', 'E' ]
console.log(arr3); // [ 'C', 'F' ]

File: RoundRobin/roundRobin.rb
# Defining the arrays
array1 = []
array2 = []
array3 = []
# Defining a counter to keep track of which array to insert the element into
counter = 1
# Defining a hash that maps the counter to the corresponding array
arrays = {
  1 => array1,
  2 => array2,
  3 => array3
}
# Defining a method to insert an element into the arrays using round-robin
def insert_element(element, counter, arrays)
  # Gettting the array to insert the element into
  array = arrays[counter]
  # Inserting the element into the array
  array << element
  # Incrementing the counter
  counter = counter % 3 + 1
  # Returning the updated counter
  return counter
end
# Inserting elements A, B, C, D, E, F into the arrays using round-robin
counter = insert_element("A", counter, arrays)
counter = insert_element("B", counter, arrays)
counter = insert_element("C", counter, arrays)
counter = insert_element("D", counter, arrays)
counter = insert_element("E", counter, arrays)
counter = insert_element("F", counter, arrays)
# Printing the arrays
puts "Array 1: #{array1.inspect}"
puts "Array 2: #{array2.inspect}"
puts "Array 3: #{array3.inspect}"